### ISSUE TITLE
`PLE1520` false positive when using `singledispatchmethod` with `staticmethod`

### ISSUE THREAD

**Issue by ForeverProglamer** [astral-sh/ruff#10619]
**Ruff version:** ruff 0.3.4.
**Python version:** Python 3.12.2.
**Code to reproduce a bug:**
```python
from functools import singledispatch, singledispatchmethod


class Foo:
    @singledispatchmethod
    @staticmethod
    def bar(value):
        raise NotImplementedError() 

    @bar.register
    @staticmethod
    def _(value: int) -> int:
        return 42 * value

    def method(self) -> int:
        return self.bar(2)


print(Foo().method())
```
**Command invoked to lint the code:** `ruff check --isolated --preview --select=PL main.py`.

```
main.py:5:5: PLE1520 `@singledispatchmethod` decorator should not be used on non-method functions
  |
4 | class Foo:
5 |     @singledispatchmethod
  |     ^^^^^^^^^^^^^^^^^^^^^ PLE1520
6 |     @staticmethod
7 |     def bar(value):
  |
  = help: Replace with `@singledispatch`
```

Running this code outputs `84`. However, if you replace `@singledispatchmethod` with `@singledispatch` as Ruff suggests, and then run the code, you will get an error:
```Traceback (most recent call last):
  File "/home/ebubuntu/projects/test/ruff/main.py", line 19, in <module>
    print(Foo().method())
          ^^^^^^^^^^^^^^
  File "/home/ebubuntu/projects/test/ruff/main.py", line 16, in method
    return self.bar(2)
           ^^^^^^^^^^^
  File "/home/ebubuntu/.pyenv/versions/3.12.2/lib/python3.12/functools.py", line 909, in wrapper
    return dispatch(args[0].__class__)(*args, **kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: Foo.bar() takes 1 positional argument but 2 were given
```

[Python docs on `singledispatchmethod` states](https://docs.python.org/3/library/functools.html#functools.singledispatchmethod):
> @singledispatchmethod supports nesting with other decorators such as [@classmethod](https://docs.python.org/3/library/functions.html#classmethod). ... The same pattern can be used for other similar decorators: [@staticmethod](https://docs.python.org/3/library/functions.html#staticmethod), [@abstractmethod](https://docs.python.org/3/library/abc.html#abc.abstractmethod), and others.

**Comment by AlexWaygood** [comment]
This issue is interesting because our behaviour is currently consistent with pylint's here. However, we discussed this internally, and we agree that the rule is indeed behaving in quite an odd way right now -- we think both pylint's behaviour and ruff's behaviour are both incorrect currently.
